transport/http/outbound.go

```diff
@@ -399,6 +405,11 @@ func getYARPCErrorFromResponse(response *http.Response, bothResponseError bool)
 	)
 }
 
+func checkServiceMatch(reqSvcName string, resHeaders *http.Header) (bool, string) {
+	svcHeaderValue := resHeaders.Get(ServiceHeader)
+	return svcHeaderValue == "" || resHeaders.Get(ServiceHeader) == reqSvcName, resHeaders.Get(ServiceHeader)
```

Comment:
since we captured the variable above, let's use it in the other two instances

Reply:
Resolved.